cookbooks/cdo-repository/libraries/git.rb

```diff
@@ -0,0 +1,62 @@
+# Patch Chef::Provider::Git to support creating/updating shallow clones.
+class Chef
+  class Provider
+    class Git < Chef::Provider
+
+      def clone
```

Comment:
Patching could be fragile when we update chef.  Is it absolutely critical to do this or could we just define a new method that has the desired behavior?